### PR TITLE
Portkey AI Integration with Groq

### PR DESCRIPTION
**Pull request by siddharthsambharia-portkey**
This PR adds documentation for integrating Portkey AI with Groq's LLMs. Portkey AI offers an AI Gateway and Observability Suite, enabling teams to build reliable and efficient AI applications.

Key Features

1. Enhanced observability with 42+ metrics and logs
2. Advanced routing features (retries, fallbacks, load balancing)
3. Semantic caching for cost optimization
4. Guardrails for input/output verification

### CODE REVIEW

File: tutorials/Portkey-with-Groq/Readme.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+```
+
+
+
+### Observability
+
+Portkey automatically logs all requests, making debugging and monitoring simple. View detailed logs and traces in the Portkey dashboard.
+
+
+![observaibility dashbaord](https://raw.githubusercontent.com/siddharthsambharia-portkey/Portkey-Product-Images/refs/heads/main/Portkey-Dashboard.png)
```
Comment (hozen-groq) [inline tutorials/Portkey-with-Groq/Readme.md:71]:
```suggestion
![observability dashboard](https://raw.githubusercontent.com/siddharthsambharia-portkey/Portkey-Product-Images/refs/heads/main/Portkey-Dashboard.png)
```

File: tutorials/Portkey-with-Groq/Readme.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+
+![image](https://raw.githubusercontent.com/siddharthsambharia-portkey/Portkey-Product-Images/refs/heads/main/Portkey-Traces.png)
+
+
+### Using 250+ Models
+
+One of Portkey's strengths is the ability to easily switch between different LLM providers. To use OpenAI instead of Groq, simply change the virtual key:
```
Comment (hozen-groq) [inline tutorials/Portkey-with-Groq/Readme.md:79]:
```suggestion
One of Portkey's strengths is the ability to easily switch between different LLM providers. To do so, simply change the virtual key.
```

File: tutorials/Portkey-with-Groq/Readme.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+```python
+portkey = Portkey(
+    api_key="YOUR_PORTKEY_API_KEY",
+    virtual_key="YOUR_OPENAI_VIRTUAL_KEY",
```
Comment (hozen-groq) [inline tutorials/Portkey-with-Groq/Readme.md:84]:
```suggestion
    virtual_key="VIRTUAL_KEY",
```

File: tutorials/Portkey-with-Groq/Readme.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+)
+```
+
+
+### Add Custom-data to your requests
```
Comment (hozen-groq) [inline tutorials/Portkey-with-Groq/Readme.md:89]: Could you please add a 1-2 sentence explanation here?

**Comment by siddharthsambharia-portkey** [comment]
@hozen-groq  Regarding your last comment, I believe it would be more fitting to place this in `/tutorials`, especially since similar cookbooks like Portkey AI already reside there. But if you insist I can change the location to `/guides`

**Comment by siddharthsambharia-portkey** [comment]
hey any updates @hozen-groq ?

**Comment by hozen-groq** [comment]
@siddharthsambharia-portkey Thank you for the updates! At a second glance, I agree with your comment about it living in `tutorials`. Just merged. 🤩

**Pull request merged**